arangod/Replication2/ReplicatedState/UnconfiguredStateManager.tpp

```diff
@@ -0,0 +1,89 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2022-2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Replication2/ReplicatedState/UnconfiguredStateManager.h"
+
+#include "Replication2/Exceptions/ParticipantResignedException.h"
+
+namespace arangodb::replication2::replicated_state {
+
+template<typename S>
+UnconfiguredStateManager<S>::UnconfiguredStateManager(
+    std::shared_ptr<ReplicatedState<S>> const& parent,
+    std::shared_ptr<replicated_log::LogUnconfiguredParticipant>
+        unconfiguredParticipant,
+    std::unique_ptr<CoreType> core, std::unique_ptr<ReplicatedStateToken> token)
+    : _parent(parent),
+      _unconfiguredParticipant(std::move(unconfiguredParticipant)),
+      _core(std::move(core)),
+      _token(std::move(token)) {}
+
+template<typename S>
+void UnconfiguredStateManager<S>::run() {
+  // We need to drop the lock before calling `thenFinal`, because it might
+  // immediately be executed and try to acquire the lock again.
+  // Thus execute it in the DeferredAction.
```

Comment:
This comment is pretty confusing here. I know what it means because I saw it yesterday before you moved the code. But future-me will just be confused.

Reply:
Agreed, will remove it.